backend/wallet/storage/transaction.py

```diff
@@ -120,6 +125,10 @@ def get_transaction_by_details(
     ).first()
 
 
+def get_transaction_by_referenced_event(referenced_event: int) -> Transaction:
+    return Transaction.query.filter_by(referenced_event=referenced_event).first()
```

Comment:
what we should do when there is duplicated txn reference event?

Reply:
reference event sequence number is not global uniq, it requires related event_key to become uniq, so we should store event_key + event_sequence_number.